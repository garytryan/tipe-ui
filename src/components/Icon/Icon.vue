<template>
  <svg
    :data-tipe-ui="$options.name"
    :style="styleObject"
    :viewBox="src.viewBox"
    :color="color"
  >
    <use :xlink:href="src.id" />
  </svg>
</template>

<script>
import vueTypes from 'vue-types'
import iconLoader from '@/libs/iconLoader'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string.def('plus'),
    height: vueTypes.string.def('1rem'),
    width: vueTypes.string.def('1rem'),
    color: vueTypes.string.def('#000'),
    __iconLoader__: vueTypes.func.def(iconLoader)
  },
  computed: {
    src() {
      const { viewBox, id } = this.__iconLoader__(this.icon)
      return {
        viewBox,
        id: `#${id}`
      }
    },
    styleObject() {
      return {
        height: this.height,
        width: this.width
      }
    }
  }
}
</script>
