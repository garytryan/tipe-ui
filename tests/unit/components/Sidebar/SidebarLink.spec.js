import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import SidebarLink from '@/components/Sidebar/SidebarLink.vue'
import Icon from '@/components/Icon'
import faker from 'faker'

describe('Sidebar', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const wrapper = shallowMount(SidebarLink)

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('<root> - has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(SidebarLink)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeSidebarLink')
    })

    it('<root> - should be a tag', () => {
      const wrapper = shallowMount(SidebarLink)

      expect(wrapper.is('a')).toBe(true)
    })

    it('<tipe-icon> - should exist', () => {
      const wrapper = shallowMount(SidebarLink)

      expect(wrapper.find(Icon).exists()).toBe(true)
    })
  })

  describe(':props', () => {
    it(':url - should default to #', () => {
      const wrapper = shallowMount(SidebarLink)

      expect(wrapper.attributes().href).toBe('#')
    })

    it(':url - should match href', () => {
      const propsData = { url: faker.internet.url() }
      const wrapper = shallowMount(SidebarLink, { propsData })

      expect(wrapper.attributes().href).toBe(propsData.url)
    })

    it(':icon - should be passed to Icon component', () => {
      const propsData = { icon: faker.lorem.word() }
      const wrapper = shallowMount(SidebarLink, { propsData })

      expect(wrapper.find(Icon).props().icon).toBe(propsData.icon)
    })
  })
})
