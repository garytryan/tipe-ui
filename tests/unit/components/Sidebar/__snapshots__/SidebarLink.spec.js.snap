// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar <template> matches previous snapshot 1`] = `
<a data-tipe-ui="TipeSidebarLink" href="#">
  <div class="icon-container">
    <TipeIcon-stub class="icon"></TipeIcon-stub>
  </div>
  <div class="indicator"></div>
</a>
`;
